opsgenie/data_source_opsgenie_service.go

```diff
@@ -0,0 +1,88 @@
+package opsgenie
+
+import (
+	"context"
+	"errors"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/opsgenie/opsgenie-go-sdk-v2/service"
+	"log"
+	"strconv"
+	"strings"
+)
+
+func dataSourceOpsGenieService() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceOpsGenieServiceRead,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateOpsGenieServiceName,
+			},
+			"team_id": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateOpsGenieServiceTeamId,
+			},
+			"description": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateOpsGenieServiceDescription,
+			},
+		},
+	}
+}
+
+func dataSourceOpsGenieServiceRead(d *schema.ResourceData, meta interface{}) error {
+	client, err := service.NewClient(meta.(*OpsgenieClient).client.Config)
+	if err != nil {
+		return err
+	}
+	name := d.Get("name").(string)
+
+	log.Printf("[INFO] Reading OpsGenie service '%s'", name)
+
+	breakFlag := false
+	offset := 0
+	paginationError := errors.New("")
```

Comment:
we dont need this definition.
we can use 		
```
offset, err = strconv.Atoi(offsetString)
if err!=nil{
return err
}
```
in the 83.line

Reply:
Changed variable name and removed object creation.